src/Microsoft.TemplateEngine.Cli/HelpAndUsage/HelpForTemplateResolution.cs

```diff
@@ -319,32 +319,33 @@ private static void DisplayPartialNameMatchLanguageAndContextProblems(string tem
             if (templateResolutionResult.IsNoTemplatesMatchedState || templateResolutionResult.UsingPartialMatches)
             {
                 ShowNoTemplatesFoundMessage(templateName, templateLanguage, context);
+                Reporter.Error.WriteLine();
                 return;
             }
 
             bool anythingReported = false;
-
-            foreach (IReadOnlyList<ITemplateMatchInfo> templateGroup in templateResolutionResult.ContextProblemMatchGroups)
-            {
-                // all templates in a group should have the same context & name
-                if (templateGroup[0].Info.Tags != null && templateGroup[0].Info.Tags.TryGetValue("type", out ICacheTag typeTag))
+            int partialTemplatesMatchCount = templateResolutionResult.ContextProblemMatchGroups.Count(templateGroup =>
                 {
-                    MatchInfo? matchInfo = WellKnownSearchFilters.ContextFilter(context)(templateGroup[0].Info);
-                    if ((matchInfo?.Kind ?? MatchKind.Mismatch) == MatchKind.Mismatch)
+                    // all templates in a group should have the same context & name
+                    if (templateGroup[0].Info.Tags != null && templateGroup[0].Info.Tags.TryGetValue("type", out ICacheTag typeTag))
                     {
-                        // {0} matches the specified name, but has been excluded by the --type parameter. Remove or change the --type parameter to use that template
-                        Reporter.Error.WriteLine(string.Format(LocalizableStrings.TemplateNotValidGivenTheSpecifiedFilter, templateGroup[0].Info.Name).Bold().Red());
-                        anythingReported = true;
+                        MatchInfo? matchInfo = WellKnownSearchFilters.ContextFilter(context)(templateGroup[0].Info);
+                        return ((matchInfo?.Kind ?? MatchKind.Mismatch) == MatchKind.Mismatch);
                     }
-                }
-                else
-                {
+
                     // this really shouldn't ever happen. But better to have a generic error than quietly ignore the partial match.
-                    //
-                    //{0} cannot be created in the target location
+                    // Cannot retrieve the type for {0}.
                     Reporter.Error.WriteLine(string.Format(LocalizableStrings.GenericPlaceholderTemplateContextError, templateGroup[0].Info.Name).Bold().Red());
                     anythingReported = true;
-                }
+                    return false;
+                });
+
+            if (partialTemplatesMatchCount > 0)
+            {
+                ShowNoTemplatesFoundMessage(templateName, templateLanguage, context);
+                // {0} template(s) partially matched, but failed on {1}.
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.TemplatesNotValidGivenTheSpecifiedFilter, partialTemplatesMatchCount, string.Concat("type=", context)).Bold().Red());
```

Comment:
nit/observation: There are many occurrences of the pattern `Reporter.Error.WriteLine(string.Format(...` in the code base so it would be worth adding an overload to `Microsoft.TemplateEngine.Cli` like so:

```C#
public void WriteLine(string format, params object[] arg)
```

(Just a random idea, out of the scope of this PR for sure.)